app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt

```diff
@@ -249,14 +168,47 @@ object FirebaseDB {
         }
     }
 
+    override suspend fun storeBinLocation(location: LatLng): Boolean {
+        val freshId = databaseReference.child(BIN_LOCATION_PATH).push().key ?: return false
+        databaseReference.child(BIN_LOCATION_PATH).child(freshId).setValue(location).await()
+        return true
+    }
+
+    override suspend fun getAllBinLocations(): Set<LatLng>? {
+        // Looks like manually deleting the unchecked cast is the only way to remove the warnings
+        @Suppress("UNCHECKED_CAST")
+        val childrens = databaseReference.child(BIN_LOCATION_PATH).get().await().value
+                as? Map<String, Any> ?: return null
+
+        val locations = mutableSetOf<LatLng>()
+        for (child in childrens.values) {
+            // Converts the Map back to a LatLng. If some child does not have the correct format, skip it
+            @Suppress("UNCHECKED_CAST")
+            val location = child as? HashMap<String, Double>? ?: child as? HashMap<String, Long>? ?: continue

```

Comment:
this is really weird from a design perspective

Reply:
Indeed I totaly agree with you, we will try to improve that.